gax/src/test/java/com/google/api/gax/batching/BatcherImplTest.java

```diff
@@ -711,6 +712,134 @@ public void run() {
     }
   }
 
+  @Test
+  public void testConstructors() throws InterruptedException {
+    BatcherImpl batcher1 =
+        new BatcherImpl<>(
+            SQUARER_BATCHING_DESC_V2,
+            callLabeledIntSquarer,
+            labeledIntList,
+            batchingSettings,
+            EXECUTOR);
+    try {
+      assertThat(batcher1.getFlowController()).isNotNull();
+      assertThat(batcher1.getFlowController().getLimitExceededBehavior())
+          .isEqualTo(batchingSettings.getFlowControlSettings().getLimitExceededBehavior());
+      assertThat(batcher1.getFlowController().getMaxOutstandingElementCount())
+          .isEqualTo(batchingSettings.getFlowControlSettings().getMaxOutstandingElementCount());
+      assertThat(batcher1.getFlowController().getMaxOutstandingRequestBytes())
+          .isEqualTo(batchingSettings.getFlowControlSettings().getMaxOutstandingRequestBytes());
+    } finally {
+      batcher1.close();
+    }
+
+    FlowController flowController =
+        new FlowController(
+            FlowControlSettings.newBuilder()
+                .setLimitExceededBehavior(LimitExceededBehavior.ThrowException.ThrowException)
+                .setMaxOutstandingRequestBytes(6000L)
+                .build());
+    BatcherImpl batcher2 =
+        new BatcherImpl<>(
+            SQUARER_BATCHING_DESC_V2,
+            callLabeledIntSquarer,
+            labeledIntList,
+            batchingSettings,
+            EXECUTOR,
+            flowController);
+    try {
+      assertThat(batcher2.getFlowController()).isSameInstanceAs(flowController);
+    } finally {
+      batcher2.close();
```

Comment:
I think you can use try with resources for this since Batcher implements AutoCloseable